### ISSUE TITLE
Increase 50hz and 60hz refresh rate logic to avoid image quality issues

### ISSUE THREAD

**Issue by atrus05** [plexinc/plex-media-player#415]
Today 50hz and 60hz is not preferred over 25hz and 30hz. This might have an impact on quality if motion enhancement features are enabled in the users TV sets. So these are my suggested changes:

* Increase weighting of 50hz and 60hz to make it become higher weighted than 25hz and 30hz if a specific setting has been set (see below).
* In mpv.conf or plexmediaplayer.conf create an advanced setting like "Prefer 50/60hz even if 25/30 exists in EDID". Set it to default Yes. The reasoning behind it is that it is a higher ratio of non-purists that uses motion enhancements as on in their TVs. And I suspect non-purists have a higher threshold to change information in hidden configuration files.
* Lastly; change PMP so it does not switch refresh rate if it is already using a multiple integer of the video FPS.

**Comment by wm4** [comment]
I'm still a bit hesitant about this.
- I still don't completely understand the purpose of this. Native refresh rate should be best.
- I can understand that for 30fps video on a 60hz you'd want to avoid the possibly annoying mode switches. So we could not switch if the current rate is an integer-multiple, as you've mentioned in your last point.
- One user in the forums mentioned that he can't adjust audio delay at 25Hz as a reason for switching to 50H. Well, maybe we should add that?
- If we prefer 50/60Hz over 25/30, should we make 50 and 60Hz exceptions, or just always prefer the highest refresh rate that's an integer-multiple? Should the same apply to 23.976Hz, so that we'd prefer 48, 72, 120Hz? (I don't like exceptions.)

**Comment by lqvnguyen** [comment]
@wm4 I am adding to this GHI for an additional reason to support the preference of 50/60Hz refresh rate logic over 25/30Hz. For the sake of this simplicity, I will list the rounded values instead of 29.97 and 59.94 in the points below. When both Display Mode Switching and Deinterlaced are enabled:

- In the case of mixed content (containing both progressive film & interlaced video) and pure interlaced, the playback of content in a MKV remuxed from a DVD is slightly unsmooth to very choppy.

- The yadif 2x filter is applied to the 60i fields and effectively 60p frames are created.

- With the refresh rate being switched to 30Hz due to the framerate reported by the MKV container, half the frames are dropped. This creates not only unsmooth video playback but noticeable lip-sync issues as well.

- I can not find a good usage case for switching to 30Hz. I have never come across a DVD player that would switch to 30hz. They always output @60Hz. I think PMP should take this same approach for content containers such as MKV that report 30FPS

- Even if there is a need to play true 30p content, leaving it at 60Hz would work without detriment as it is a interger-multiple.

- For purely progressive telecined film content (24p), my opinion is that the picture playback is smoother when 3:2 progressive pulldown is applied for 60hz instead of whatever progressive pulldown ratio is applied for 30hz.

I know I have been discussing this with you on the forums for a year. I was finally able to wrap my head around the subject matter to clearly convey the issue. I think you can resolve the issue by either:

1. Removing 30Hz as a valid refresh rate that can be selected by the player.
2. As @atrus05 suggested, maybe add code that can only be configured as an advanced setting in plexmediaplayer.conf

In my opinion refresh rate switching brings more detriment than good for 30Hz content.

**Comment by lqvnguyen** [comment]
@wm4 I know you don't like exceptions but I think 23.976fps content needs to switch to 23.976Hz. This is required for most TVs to display at 2:2,3:3,4:4,5:5 and 10:10 cadence.

**Comment by wm4** [comment]
>I think 23.976fps content needs to switch to 23.976Hz. 

It should already do this.

**Comment by lqvnguyen** [comment]
Sorry--let me rephrase:
> Should the same apply to 23.976Hz, so that we'd prefer 48, 72, 120Hz? (I don't like exceptions.)
I know you don't like exceptions but I think 23.976fps content needs to switch to 23.976Hz and not prefer 48, 72, 120Hz.

**Comment by wm4** [comment]
Ah, and why not?

**Comment by lqvnguyen** [comment]
> Ah, and why not?

If you want most displays to repeat the 23.976fps to 48fps,72fps,96fps,120,fps,240fps you have to output a refresh of 23.976Hz. The primary purpose for the concept of 24p input on a display was derived from BD players outputting a true 24p signal to the display so that it can avoid motion judder from 3:2 pulldown. 

1. The HDTV displays do one of two things when feed a 24p signal
- Repeat the frames using 2:2,3:3,4:4,5:5 and 10:10 cadence for 48Hz, 72Hz, 96Hz, 120Hz and 240Hz--depending on the refresh rates the panel/projector is capable of. (Plasma panels and Projectors can usually run at more than one refresh rate. However LCD panels typical have to run at their native refresh rates)
- Apply Motion Interpolation to create artificial frames instead of repeating the frames to hit the desired/native refresh rate.

2. There would be no point in trying to switch to anything higher. The majority of displays on the market can not accept input on anything higher than 60Hz.

3. 23.976Hz is the only refresh rate that is a divisible integer for the other higher rates.

**Comment by wm4** [comment]
Here, I added a setting that skips 25 and 30 Hz (and close) modes. The autoswitching code should naturally prefer the next multiples if the setting is enabled. It's not exposed in the UI.

> Apply Motion Interpolation to create artificial frames instead of repeating the frames to hit the desired/native refresh rate.

That's the only real point. Depends whether you like this sort of interpolation. The only legitimate reason to do interpolation is when you don't have an exact refresh rate (or close multiple) in the first place.

> 23.976Hz is the only refresh rate that is a divisible integer for the other higher rates.

Doesn't matter. The speed change is barely noticeable.

**Comment by LongChair** [comment]
@atrus05 : is the last fix Vincent mentionned enough to close this or do you consider this still being an issue ?
